### PR TITLE
sdk format - preserve comments

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24046

## Description
Moved to ruamel instead of yaml.

## Must have
- [x] Tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Changelog
+* Fixed a bug in the **format** command where comments were being removed.
```
Comment (Itay4) [inline CHANGELOG.md:2]:
```suggestion
* Fixed a bug in the **format** command where comments were being removed from YAML files.
```

File: demisto_sdk/commands/common/tools.py [hunk 2]
```diff
@@ -576,7 +592,7 @@ def get_dev_requirements(py_version, envs_dirs_base):
     return requirements
 
 
-def get_dict_from_file(path: str) -> Tuple[Dict, Union[str, None]]:
+def get_dict_from_file(path: str, use_ryaml: bool = False) -> Tuple[Dict, Union[str, None]]:
     """
     Get a dict representing the file
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py:604]: add arg to docstring

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -281,6 +286,17 @@ def get_yaml(file_path):
     return get_file(yaml.safe_load, file_path, ('yml', 'yaml'))
 
 
+def get_ryaml(file_path: str) -> dict:
+    """
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py:290]: use google style docstring

File: demisto_sdk/commands/common/tools.py [hunk 4]
```diff
@@ -281,6 +286,23 @@ def get_yaml(file_path):
...
+    Get yml file contents using ruaml
+    :param file_path: The file path
+    :return: The yml contents
+    """
+    try:
+        with open(os.path.expanduser(file_path), 'r') as yf:
+            data = ryaml.load(yf)
+    except FileNotFoundError as e:
+        print_error(f'File {file_path} not found. Error wasL {str(e)}')
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py:299]: what is `wasL` ?

Comment (Itay4) [inline demisto_sdk/commands/common/tools.py:299]: we should get rid of `print_error`, but use `click.echo`

File: demisto_sdk/commands/format/tests/test_formatting_yml_test.py [hunk 5]
```diff
@@ -23,6 +29,16 @@
 ]
 
 
+@pytest.mark.parametrize('source_path, destination_path, formatter, yml_title, file_type', BASIC_YML_TEST_PACKS)
+def test_yml_preserve_comment(source_path, destination_path, formatter, yml_title, file_type, capsys):
```
Comment (Itay4) [inline demisto_sdk/commands/format/tests/test_formatting_yml_test.py:33]: add test docstring `Given-When-Then`

File: demisto_sdk/commands/common/tools.py (hunk 4 above)
Comment (guyfreund) [reply demisto_sdk/commands/common/tools.py:299]: should be `:`

File: demisto_sdk/commands/common/tools.py [hunk 6]
```diff
@@ -12,9 +12,10 @@
 from subprocess import DEVNULL, PIPE, Popen, check_output
 from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
-import git
+import click
+import git  # type: ignore
 import requests
-import urllib3
+import urllib3  # type: ignore
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py]: what
why

File: demisto_sdk/commands/common/tools.py [hunk 7]
```diff
@@ -12,9 +12,10 @@
 from subprocess import DEVNULL, PIPE, Popen, check_output
 from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
-import git
+import click
+import git  # type: ignore
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py]: what
why

**Pull request merged**